### PR TITLE
Development

### PR DESCRIPTION
**Pull request by latoyazamill**
## Completed on Tuesday

- [x] Create a findToDo database function
- [x] Create an updateTodo database function 
- [x] Create a sign-in GET route
- [x] Create a sign-up GET route
- [x] Create a sign-up POST route
- [x] Create a to do list form 
- [x] Create a sign-in POST route

I implemented your changes from yesterday. The only issues that I am noticing thus far today is with my `/signin` route. I am getting an error message even with a valid password. Other than that, all other feedback is welcomed! Thanks so much :)

### CODE REVIEW

File: src/models/db/users.js [hunk 1]
```diff
@@ -1,27 +1,52 @@
...
+};
+
+const signIn = (email, password) => {
+  return findByEmail(email)
+  .then((user) => {
+    return bcrypt.compare(password, user.password)
+    .then((validUser) => {
+      if (!validUser) {
+        throw new Error ('Invalid Username/Password')
+      }
```
Comment (rae-ralston) [inline src/models/db/users.js:46]: do we need to return validUser here? I imagine it's working or it wouldn't be in your PR...?

File: src/server/routes/index.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  response.render('todo/todo-list');
+});
+
+router.get('/signup', (request, response) => {
+  response.render('users/sign-up');
+});
+
+router.post('/signup', (request, response) => {
+  users.signUp(request.body)
+  .then((user) => {
```
Comment (rae-ralston) [inline src/server/routes/index.js]: indent this `then` block

Comment (rae-ralston) [reply src/server/routes/index.js]: also should you do some check here? if `user` is true would go to profile page? What happens if user didn't get signed up here?

File: src/server/routes/index.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+});
+
+router.get('/signin', (request, response) => {
+  response.render('users/sign-in');
+});
+
+router.post('/signin', (request, response) => {
+  const { email, password } = request.body
+  users.signIn(email, password)
+  .then((user) => {
```
Comment (rae-ralston) [inline src/server/routes/index.js]: indent `then` block

File: src/views/todo/todo-list.ejs [hunk 4]
```diff
@@ -2,4 +2,19 @@
...
+  <input type="text" name="item" required>
+
+  <input type="submit" name="to-do-list-button" value="Submit">
+
+</form>
+
+<h2>List of To Dos:</h2>
+
+<ol>
+  <li> Example Text<br><input type="radio" name="to-do-list-item"> Complete</li>
```
Comment (rae-ralston) [inline src/views/todo/todo-list.ejs]: should this be a radio button or a check box? haven't seen the UI, just a thought.

File: src/models/db/users.js [hunk 5]
```diff
@@ -1,27 +1,52 @@
...
       *
-  `, [user.first_name, user.email, user.password])
+  `, [user.full_name, user.email, user.password])
   .catch(console.log)
 };
 
-const signIn = (email, password) => {
+const signUp = (user) => {
+    return encryptPassword(user.password).then((hash) => {
+      user.password = hash
```
Comment (rae-ralston) [inline src/models/db/users.js]: be consistent about formatting promise chains. Do you line break for the `.then` or not?

**Review by rae-ralston — COMMENTED** [review]
Overall, still looking great. Not seeing any big problems, see comments for what I did see. So glad you're doing this! <3

File: src/models/db/users.js (hunk 5 above)
Comment (latoyazamill) [reply src/models/db/users.js]: Complete 👍

File: src/models/db/users.js (hunk 1 above)
Comment (latoyazamill) [reply src/models/db/users.js:46]: Complete 👍

File: src/server/routes/index.js (hunk 2 above)
Comment (latoyazamill) [reply src/server/routes/index.js]: Complete 👍
And the fields are required. If the user does not signup then nothing happens. They would need to sign up. What I can do is add a link to the form to send the user to the signin page if they need to signin instead. When I user logs in, they will immediately see the todo list form and their todos.

File: src/server/routes/index.js (hunk 3 above)
Comment (latoyazamill) [reply src/server/routes/index.js]: Complete 👍

File: src/views/todo/todo-list.ejs (hunk 4 above)
Comment (latoyazamill) [reply src/views/todo/todo-list.ejs]: Complete 👍

**Pull request merged**